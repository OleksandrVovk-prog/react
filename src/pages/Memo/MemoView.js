import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import MemoComponent from './components/MemoComponent';
import Button from '../../components/Button/Button';
import styles from './sass/Memo.module.scss';

export default function MemoView({
  title, production, poster, handleClick, clickCounter, changeTitle
}) {
  const { t } = useTranslation();
  return (
    <div className={styles.memo}>
      <MemoComponent
        title={title}
        production={production}
        poster={poster}
      />
      <div className={styles.events}>
        <p className={styles.update}>
          {t('likes')}
          &nbsp;
          {clickCounter}
        </p>
        <Button
          title={t('like')}
          buttonClass={`${styles.update} ${styles.updateBtn}`}
          handleClick={handleClick}
          id="click-handler-button"
        />
        <input
          type="text"
          className={styles.update}
          value={title}
          onChange={changeTitle}
          data-testid="change-title-input"
        />
      </div>
    </div>
  );
}

MemoView.propTypes = {
  title: PropTypes.string.isRequired,
  production: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  clickCounter: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
  changeTitle: PropTypes.func.isRequired,
};
