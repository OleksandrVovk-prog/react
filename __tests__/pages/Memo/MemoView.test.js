import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../settings/test-utils';
import MemoView from '../../../src/pages/Memo/MemoView';

describe('<MemoView />', () => {
  let wrapper;
  const mockString = 'mock string';
  const mockNumber = 1;
  const mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <MemoView
        title={mockString}
        production={mockString}
        poster={mockString}
        clickCounter={mockNumber}
        handleClick={mockFunc}
        changeTitle={mockFunc}
      />
    );
  });

  test('should render MemoView component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should call mockFunc', async () => {
    const { getByTestId } = wrapper;
    const Input = getByTestId(/change-title-input/i);
    await userEvent.type(Input, 'change input handler');
    expect(mockFunc).toHaveBeenCalled();
  });
});
