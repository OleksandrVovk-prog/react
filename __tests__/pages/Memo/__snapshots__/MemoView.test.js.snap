// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoView /> should render MemoView component 1`] = `
<DocumentFragment>
  <div
    class="memo"
  >
    <div
      class="info"
    >
      <p
        class="appIntro"
      >
        Title: mock string
      </p>
      <p
        class="appIntro"
      >
        Production: mock string
      </p>
      <img
        alt="mock string"
        class="appBanner"
        src="mock string"
      />
    </div>
    <div
      class="events"
    >
      <p
        class="update"
      >
        Likes: 1
      </p>
      <button
        class="update updateBtn"
        data-testid="click-handler-button"
        type="button"
      >
        Like
      </button>
      <input
        class="update"
        data-testid="change-title-input"
        type="text"
        value="mock string"
      />
    </div>
  </div>
</DocumentFragment>
`;
